umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -0,0 +1,848 @@
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/sentry"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/httpclient"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+
+	dto "github.com/prometheus/client_model/go"
+	redpandayaml "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/redpandaserviceconfig"
+	s6fsm "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/fsm/s6"
+	s6service "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/s6"
+)
+
+// IRedpandaService is the interface for managing Redpanda
+type IRedpandaService interface {
+	// GenerateS6ConfigForRedpanda generates a S6 config for a given redpanda instance
+	GenerateS6ConfigForRedpanda(redpandaConfig *redpandaserviceconfig.RedpandaServiceConfig) (s6serviceconfig.S6ServiceConfig, error)
+	// GetConfig returns the actual Redpanda config from the S6 service
+	GetConfig(ctx context.Context) (redpandaserviceconfig.RedpandaServiceConfig, error)
+	// Status checks the status of a Redpanda service
+	Status(ctx context.Context, tick uint64) (ServiceInfo, error)
+	// AddRedpandaToS6Manager adds a Redpanda instance to the S6 manager
+	AddRedpandaToS6Manager(ctx context.Context, cfg *redpandaserviceconfig.RedpandaServiceConfig) error
+	// UpdateRedpandaInS6Manager updates an existing Redpanda instance in the S6 manager
+	UpdateRedpandaInS6Manager(ctx context.Context, cfg *redpandaserviceconfig.RedpandaServiceConfig) error
+	// RemoveRedpandaFromS6Manager removes a Redpanda instance from the S6 manager
+	RemoveRedpandaFromS6Manager(ctx context.Context) error
+	// StartRedpanda starts a Redpanda instance
+	StartRedpanda(ctx context.Context) error
+	// StopRedpanda stops a Redpanda instance
+	StopRedpanda(ctx context.Context) error
+	// ForceRemoveRedpanda removes a Redpanda instance from the S6 manager
+	ForceRemoveRedpanda(ctx context.Context) error
+	// ServiceExists checks if a Redpanda service exists
+	ServiceExists(ctx context.Context) bool
+	ReconcileManager(ctx context.Context, tick uint64) (error, bool)
+	// IsLogsFine checks if the logs of a Redpanda service are fine
+	// Expects logs ([]s6service.LogEntry), currentTime (time.Time), and logWindow (time.Duration)
+	IsLogsFine(logs []s6service.LogEntry, currentTime time.Time, logWindow time.Duration) bool
+	// IsMetricsErrorFree checks if the metrics of a Redpanda service are error-free
+	IsMetricsErrorFree(metrics Metrics) bool
+	// HasProcessingActivity checks if a Redpanda service has processing activity
+	HasProcessingActivity(status RedpandaStatus) bool
+}
+
+// HTTPClient interface for making HTTP requests
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// defaultHTTPClient is the default implementation of HTTPClient
+type defaultHTTPClient struct {
+	client *http.Client
+}
+
+func newDefaultHTTPClient() *defaultHTTPClient {
```

Comment:
replace this with my new implementation with httpclient and the better timeout logic